doc: Document the JSON logging functionality in the proxy (#3667)
kroxylicious/kroxylicious: kroxylicious-docs/docs/_modules/monitoring/con-proxy-structured-log-fields.adoc

////
  // Copyright Kroxylicious Authors.
  //
  // Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
////

:_mod-docs-content-type: CONCEPT

// file included in the following:
//
// assembly-proxy-monitoring.adoc

[id='con-proxy-structured-log-fields-{context}']
= Structured log fields

[role="_abstract"]

> Kroxylicious uses structured logging throughout, attaching context as key-value pairs that are visible in both text and JSON output.

Reviewer: ```suggestion
Kroxylicious uses structured logging, attaching context as key-value pairs that are visible in both text and JSON output.
```

> In text format they appear at the end of each log line; in JSON format they appear as top-level fields.

Reviewer: ```suggestion
In text format, they appear at the end of each log line.
In JSON format, they appear as top-level fields.
```

Reviewer: Putting sentences on separate lines makes reviews easier

> The following table lists the commonly emitted fields.

Reviewer: ```suggestion
The following table lists commonly emitted fields.
```

.Structured log fields
[cols="1,3",options="header"]
|===
|Field |Description

|`sessionId`

> |Unique identifier for the client-to-proxy connection. Use this to correlate all log messages for a single client session.

Reviewer: ```suggestion
|Unique identifier for the client-to-proxy connection. Use this field to correlate all log messages for a single client session.
```

|`clientId`
|The Kafka `client.id` value supplied by the connecting Kafka client.

|`apiKey`

> |The Kafka API key identifying the request type (for example, `Produce`, `Fetch`, `Metadata`).

Reviewer: ```suggestion
|The Kafka API key that identifies the request type (for example, `Produce`, `Fetch`, `Metadata`).
```

|`apiVersion`

> |The version of the Kafka API used in the request.

Reviewer: ```suggestion
|The Kafka API version used in the request.
```

|`kekRef`

> |Key Encryption Key reference used by the record-encryption filter.

Reviewer: ```suggestion
|Reference to the Key Encryption Key (KEK) reference used by the Record Encryption filter.
```

Reviewer: Is `kekRef` always present only when the Record Encryption filter is enabled? If so, clarify that this field is emitted only when the filter is configured.

Author: This field is only emitted by the HashiCorp Vault KMS provider, not by the Record Encryption filter in general. I've updated the description to clarify this

|`error`

> |Error message associated with a failure condition.

Reviewer: Do we need to clarify scope? Is this field always populated for all error scenarios, or only for specific filters or components?

Author: This is a general-purpose field used broadly across the proxy runtime, filters, and KMS providers. I've updated the description to reflect that
